packages/gatsby/src/utils/api-node-docs.js

```diff
@@ -12,49 +12,51 @@ exports.resolvableExtensions = true
  *
  * See also [the documentation for the action `createPage`](/docs/actions/#createPage).
  * @example
+ * const path = require(`path`)
+ *
  * exports.createPages = ({ graphql, actions }) => {
  *   const { createPage } = actions
- *   return new Promise((resolve, reject) => {
- *     const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
- *     // Query for markdown nodes to use in creating pages.
- *     resolve(
- *       graphql(
- *         `
- *       {
- *         allMarkdownRemark(limit: 1000) {
- *           edges {
- *             node {
- *               fields {
- *                 slug
- *               }
+ *   const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
+ *   // Query for markdown nodes to use in creating pages.
+ *   // You can query for whatever data you want to create pages for e.g.
+ *   // products, portfolio items, landing pages, etc.
+ *   graphql(`
+ *     {
+ *       allMarkdownRemark(limit: 1000) {
+ *         edges {
+ *           node {
+ *             fields {
+ *               slug
  *             }
  *           }
  *         }
  *       }
- *     `
- *       ).then(result => {
- *         if (result.errors) {
- *           reject(result.errors)
- *         }
- *
- *         // Create blog post pages.
- *         result.data.allMarkdownRemark.edges.forEach(edge => {
- *             createPage({
- *               path: `${edge.node.fields.slug}`, // required
- *               component: blogPostTemplate,
- *               context: {
- *                 // Add optional context data. Data can be used as
- *                 // arguments to the page GraphQL query.
- *                 //
- *                 // The page "path" is always available as a GraphQL
- *                 // argument.
- *               },
- *             })
- *         })
+ *     }
+ *   `).then(result => {
+ *     if (result.errors) {
+ *       reject(result.errors)
```

Comment:
```suggestion
  *       return Promise.reject(result.errors)
```

Reply:
Oh this should be a `throw`, yikes, didn't read this very closely.